test/tracecontext/sample_rate.js

```diff
@@ -0,0 +1,218 @@
+    agent._conf.transactionSampleRate = 0
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing')
+    const span = transaction.startSpan('foo')
+    t.equals(transaction.sample_rate, 0, 'sample rate set')
+    t.ok(!span, 'no span for unsampled transactions')
+    t.end()
+  })
+
+  // test that an unsampled transaction from a non-zero sample rate
+  // still produces a final serialized span with a sample rate of 0
+  test.test('recorded root transactions have a sample rate of 0', function (t) {
+    agent._conf.transactionSampleRate = 0.1
+
+    // sort of gross while shenanagins to let us get an unsampled
+    // transaction with a non-zero sample rate
+    const transaction = startUnSampledTransaction()
+    const span = transaction.startSpan('foo')
+    const serialized = transaction.toJSON()
+    t.equals(serialized.sample_rate, 0, 'serialized sample rate is zero')
+    t.ok(!span, 'no span for unsampled transactions')
+    t.end()
+  })
+
+  // Test that an invalid tracestate in a recorded transaction
+  // results in a serialized span without a sample_rate, per the spec.
+  test.test('blank tracestate, recorded', function (t) {
+    // sort of gross while shenanagins to let us get an unsampled
+    // transaction with a non-zero sample rate
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+      childOf: TRACEPARENT_RECORDED,
+      tracestate: 'notavalidtracestate'
+    })
+    const span = transaction.startSpan('foo')
+    span.end()
+    span._encode(function (err, spanSerialized) {
+      t.error(err)
+      const transactionSerialized = transaction.toJSON()
+
+      t.equals(transactionSerialized.sample_rate, undefined, 'serialized transaction should have no sample_rate')
+      t.equals(spanSerialized.sample_rate, undefined, 'serialized span should have no sample_rate')
+
+      t.end()
+    })
+  })
+
+  // Test that a blank tracestate in a recorded transaction
+  // results in a serialized span without a sample_rate, per the spec.
+  test.test('blank tracestate, recorded', function (t) {
+    // sort of gross while shenanagins to let us get an unsampled
+    // transaction with a non-zero sample rate
+    const transaction = agent.startTransaction('foo', 'bar', 'baz', 'bing', {
+      childOf: TRACEPARENT_RECORDED,
+      tracestate: ''
+    })
+    const span = transaction.startSpan('foo')
+    span.end()
+    span._encode(function (err, spanSerialized) {
+      t.error(err)
+      const transactionSerialized = transaction.toJSON()
+
+      t.equals(transactionSerialized.sample_rate, undefined, 'serialized transaction should have no sample_rate')
+      t.equals(spanSerialized.sample_rate, undefined, 'serialized span should have no sample_rate')
+
+      t.end()
+    })
+  })
+
+  // Test that an invalid tracestate in an unrecorded transaction
+  // results in a serialized span without a sample_rate, per the spec.
+  test.test('blank tracestate, recorded', function (t) {
```

Comment:
copypasta: Update this test name to match what is tested.

Reply:
🤦 when the comment is right but the test name is 🍝 , fixed